ledger-core/virtual/execute/execute.go

```diff
@@ -543,16 +543,22 @@ func (s *SMExecute) stepSaveNewObject(ctx smachine.ExecutionContext) smachine.St
 	case requestresult.SideEffectNone:
 	case requestresult.SideEffectActivate:
 		_, class, memory = executionNewState.Activate()
-		action = s.setNewState(class, memory)
+		action = s.getUpdateStateAction(class, memory)
 	case requestresult.SideEffectAmend:
 		_, class, memory = executionNewState.Amend()
-		action = s.setNewState(class, memory)
+		action = s.getUpdateStateAction(class, memory)
 	case requestresult.SideEffectDeactivate:
 		panic(throw.NotImplemented())
 	default:
 		panic(throw.IllegalValue())
 	}
 
+	if s.migrationHappened {
```

Comment:
this should be done only if requestresult.SideEffectActivate or requestresult.SideEffectAmend
it is possible that nothing should be done with descriptor

Reply:
and test for this case should exist >_<

Reply:
"it is possible that nothing should be done with descriptor" - What do you mean? I shouldn't change s.execution.ObjectDescriptor inside 'if' statement?

Reply:
My point is that we should not send latest state if state hasn't changed.